### PR TITLE
Allow installing a module from a specific branch

### PR DESCRIPTION
**Pull request by Fripe070**
### Checklist

- [X] I have planned and discussed this feature with other contributors.
- [X] I have run basic tests on my code to ensure it is functional.
- [ ] I have thoroughly tested my code with various states, contexts and edge cases.
- [X] I have searched for a similar pull request in this repository and didn't find anything relevant.

### Summary
Allow installing a module from a specific branch through a new `branch` field in the `/module install` command

### CODE REVIEW

File: breadcord/core_modules/modulemanager/__init__.py [hunk 1]
```diff
@@ -69,10 +69,11 @@ async def cog_app_command_error(
             ))
 
     @app_commands.command(description="Install a module from github")
-    @app_commands.describe(module="A github url or repo path")
+    @app_commands.describe(module="A github url or repo path", branch="A branch of the repo")
     @app_commands.check(breadcord.helpers.administrator_check)
-    async def install(self, interaction: discord.Interaction, module: str):
-        module = GH_BASE_URL.sub('', module).removesuffix('.git')
+    async def install(self, interaction: discord.Interaction, module: str, branch: str | None = None):
+        module = GH_BASE_URL.sub('', module).removesuffix('.git').replace(" ", "-")
```
Comment (m31-galaxy) [inline breadcord/core_modules/modulemanager/__init__.py:75]: Why is this changed?

Comment (Fripe070) [reply breadcord/core_modules/modulemanager/__init__.py:75]: I’m going to assume you mean line 75. 
It was changed to be more in line with how I handle branches, which was done to give more leniency when specifying the name, and GitHub replaces spaces in repo/branch names with dashes upon creation.

Comment (m31-galaxy) [reply breadcord/core_modules/modulemanager/__init__.py:75]: Alright - I was unsure about the decision to default to hyphens instead of e.g. underscores, but if that's GitHub's default then I'm fine with that. It's probably now deserving of a transformer with autocomplete though

**Review by m31-galaxy — APPROVED** [review]

File: breadcord/core_modules/modulemanager/__init__.py (hunk 1 above)
Comment (Fripe070) [reply breadcord/core_modules/modulemanager/__init__.py:75]: Making a transformer sounds suspiciously much like a you issue :trollface:
